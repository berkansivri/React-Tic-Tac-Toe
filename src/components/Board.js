import React, { useContext } from 'react'
import Square from './Square'
import GameContext from '../context/game-context'

const Board = () => {
  const { moves, turn } = useContext(GameContext)

  const renderSquare = (i) => {
    const move = moves.find(m => m.location === i)
    return (
      <Square location={i} sign={move && move.sign}/>
    )
  }

  return (
    <div>
      <div className="board-row">
        {renderSquare(0)}
        {renderSquare(1)}
        {renderSquare(2)}
      </div>
      <div className="board-row">
        {renderSquare(3)}
        {renderSquare(4)}
        {renderSquare(5)}
      </div>
      <div className="board-row">
        {renderSquare(6)}
        {renderSquare(7)}
        {renderSquare(8)}
      </div>
      <p style={{textAlign:"center"}}>Turn: {turn}</p>
    </div>
  )
}

export { Board as default }
