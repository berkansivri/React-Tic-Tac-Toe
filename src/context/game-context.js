import React, { useContext } from 'react'

const gameContext = React.createContext()

const useGameContext = () => useContext(gameContext)

export { gameContext as default, useGameContext }
