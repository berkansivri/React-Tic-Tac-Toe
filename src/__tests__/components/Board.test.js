import React from 'react'
import { shallow } from 'enzyme'
import Board from '../../components/Board'
import * as Context from '../../context/game-context'
import moves from '../fixtures/moves'

let contextValues, wrapper

beforeEach(() => {
  contextValues = {
    dispatch: jest.fn(),
    turn: "X",
    setTurn: jest.fn(),
    moves
  }
  jest.spyOn(Context, 'useGameContext')
      .mockImplementation(() => contextValues)
  
  wrapper = shallow(<Board />)
})

test("should render board component correctly", () => {
  expect(wrapper).toMatchSnapshot()
})

test("should render all square", () => {
  expect(wrapper.find("Square")).toHaveLength(9)
  const xSignLengh = moves.filter(m=>m.sign === "X").length
  const ySignLengh = moves.filter(m=>m.sign === "Y").length
  expect(wrapper.find("Square").find({ sign:"X" })).toHaveLength(xSignLengh)
  expect(wrapper.find("Square").find({ sign:"Y" })).toHaveLength(ySignLengh)
})