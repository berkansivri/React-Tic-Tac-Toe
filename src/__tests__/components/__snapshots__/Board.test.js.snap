// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render board component correctly 1`] = `
<div>
  <div
    className="board-row"
  >
    <Square
      location={0}
      sign="X"
    />
    <Square
      location={1}
      sign="O"
    />
    <Square
      location={2}
      sign="X"
    />
  </div>
  <div
    className="board-row"
  >
    <Square
      location={3}
      sign="O"
    />
    <Square
      location={4}
      sign="X"
    />
    <Square
      location={5}
    />
  </div>
  <div
    className="board-row"
  >
    <Square
      location={6}
    />
    <Square
      location={7}
    />
    <Square
      location={8}
      sign="X"
    />
  </div>
  <p
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Turn: 
    X
  </p>
</div>
`;
